src/unix/unix_internal.h

```diff
@@ -680,8 +673,7 @@ static inline boolean validate_user_memory(const void *p, bytes length, boolean
 {
     u64 v = u64_from_pointer(p);
 
-    /* no zero page access */
-    if (v < PAGESIZE)
+    if (v < MIN(PAGESIZE, current->p->mmap_min_addr))
```

Comment:
This check is for validate_user_memory(), which should determine whether or not a range is within a valid space for user memory (architecturally, not by validating pages). In this case, the test is only to determine whether or not to consider zero-page accesses as valid, because a mmap_min_addr value of 0 enables zero-page access. The MIN with PAGESIZE is necessary because, even when mmap_min_addr is greater than PAGESIZE, accesses to memory between PAGESIZE and mmap_min_addr are still valid.

This is independent of the allocating of address space on an mmap. mmap_min_addr values larger than a page size are not ignored; they should act as the minimum allowed address when allocating space via mmap and mremap.